go-controller/pkg/clustermanager/status_manager/status_manager.go

```diff
@@ -0,0 +1,135 @@
+package status_manager
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	adminpolicybasedrouteapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/level_driven_controller"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+// fieldManagerName should be different from any zone name
+const fieldManagerName = "cluster-manager"
+
+type resourceManager[T any] interface {
+	// cluster-scoped resources should ignore namespace
+	get(namespace, name string) (*T, error)
+	// statusChanged is used to decide if an event should trigger status update
+	statusChanged(oldObj, newObj *T) bool
+	// updateStatus should handle nil in case of pointer type, and return (patchBytes, needsUpdate).
+	updateStatus(obj *T, applyOpts *metav1.ApplyOptions) error
+}
+
+type typedStatusManager[T any] struct {
+	name string
+
+	objController level_driven_controller.StoppableController[T]
```

Comment:
We are just doing start & stop so objController does not need to be typed

Reply:
yeah I could define an interface for both controller types, but it is not really "required" here, and I also prefer to not use interfaces unless there are 2 different implementations for it, since it complicates the code in my opinion.
e.g. I find https://github.com/ovn-org/ovn-kubernetes/pull/3750/files#diff-25788e003f7180dde6dbf91380e451e092a7f0b67e850209ac884b15fc4973ffR54 easier to read and navigate with the typed generic compared to an interface, not a strong opinion though :)

Reply:
I just think there should only be one.

And iI am of a different opinion :P, generics code where types are not inferred is much harder to read than interfaces.

Reply:
the type is inferred here, its same as `typedStatusManager` type.
Interface is an extra layer of abstraction, that I would say should be avoided if you can use specific type. Generic controller is like a regular controller, it doesn't require an interface by itself unless you need some controllers manager that should work with any controller type.

Reply:
I mean in general. Reading `...[...]` its never better than just `...`, whatever it is, IMHO.

Interfaces are THE abstraction that should never be avoided, also IMHO.

Anyhow, I am not pushing for the interface really. It is not that important here since the controller is just an utility. It is just that I saw this first, before seeing that you had to ways to start and stop, so I when I saw this I thought your intention was something more akin to an interface. You can ignore this one.